src/util.rs

```diff
@@ -472,7 +492,58 @@ pub(crate) mod ptr {
         }
     }
 
-    impl<'a, T: 'a + ?Sized> Debug for Ptr<'a, T> {
+    impl<'a, T: 'a + ?Sized, const ASSUME_ALIGNED: bool> Debug for Ptr<'a, T, ASSUME_ALIGNED> {
+        #[inline]
+        fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+            self.ptr.fmt(f)
+        }
+    }
+
+    /// A reference to a validly-initialized `T` of lifetime `'a` that might
+    /// *not* be validly aligned for `T`.
+    #[doc(hidden)]
+    #[allow(unreachable_pub)] // false positive?
+    pub struct MaybeAligned<'a, T: ?Sized> {
```

Comment:
Can this go in its own module? Having these methods reach into the internals of `Ptr` wigs me out given how complex its internal invariants are.